src/python/grpcio/grpc/experimental/aio/_channel.py

```
@@ -50,10 +48,11 @@ def __init__(self, channel, method, request_serializer,
         if compression:
             raise NotImplementedError("TODO: compression not implemented yet")
 
-        response = await self._channel.unary_unary(
-            self._method, _common.serialize(request, self._request_serializer))
+        serialized_request = _common.serialize(request, self._request_serializer)
+        timeout = None if timeout is None else time.time() + timeout
+        response = await self._channel.unary_unary(self._method, serialized_request, timeout)
 
-        return _common.deserialize(response, self._response_deserializer)
+        return _common.deserialize(response or b"", self._response_deserializer)
```

`response or b""` having the feeling that you have been forced to do so since a timed out call follows the same path as the happy path. Having the feeling that a timed out call should raise an exception, and if so this would not be necessary.

Whats the current behavior when a timeout is raised in the synchronous API?

---

In the synchronous API, the server calls ``.cancel()`` and returns a message with the error, stating that the deadline has been exceeded. In that case, shouldn't then this be implemented on the server, once we have the asynchronous one available? Maybe a TODO?

---

Correct me if I'm wrong but theoretically, we are using a server with full batteries included - the synchronous one, so theoretically we should be able to catch this error and return back to the user the proper explicit message telling that the call finished with a timeout.

Once we start using the Asynchronous server, the test should remain green, otherwise would mean that something wrong is implemented in the server.

Also, I'm wondering if a deadline can be only be raised by the Server. Does the client have a mechanism for telling explicitly that the timeout was reached? Considering that the network or the server can fail which will result in not telling the client that the deadline was reached.  We would need to clarify this, and if see if both use cases share the same path.

---

Regarding 

> Also, I'm wondering if a deadline can be only be raised by the Server

I didn't test it but this is what is being said in the gRPC documentation [1]

> In gRPC, both the client and server make their own independent and local determination about whether the remote procedure call (RPC) was successful. This means their conclusions may not match! An RPC that finished successfully on the server side can fail on the client side. For example, the server can send the response, but the reply can arrive at the client after their deadline has expired. The client will already have terminated with the status error DEADLINE_EXCEEDED. This should be checked for and managed at the application level.

[1] https://grpc.io/blog/deadlines/